mysql-test/suite/binlog/t/binlog_mysqlbinlog_warn_stop_position.inc

```diff
@@ -0,0 +1,115 @@
+#
+# Helper file that ensures mysqlbinlog --stop-position behavior for local
+# files or a remote server
+#
+# Parameters:
+#   use_local_files (bool): A boolean that changes which source to use for
+#     mysqlbinlog. When true, uses local files; when false, reads remotely.
+#
+
+--connection default
+--let $MYSQLD_DATADIR= `select @@datadir`
+
+# PARAM_READ_FROM_REMOTE is used as a parameter to mysqlbinlog (_OUT suffix is
+# output in echo commands). If using local files, they are blank; if reading
+# from remote server, it is overridden to the correct values.
+--let $PARAM_READ_FROM_REMOTE=
+# Used in echo statements to remove potentially changing values
+--let $PARAM_READ_FROM_REMOTE_OUT=
+
+if (!$use_local_files)
+{
+  --let $PARAM_READ_FROM_REMOTE= --read-from-remote-server --user=root --host=127.0.0.1 --port=$MASTER_MYPORT
+  --let $PARAM_READ_FROM_REMOTE_OUT= --read-from-remote-server
+  # binlog files in --read-from-remote-server don't use file system path
+  --let $binlog_f1_full= $binlog_f1
+  --let $binlog_f2_full= $binlog_f2
+}
+if ($use_local_files)
+{
+  # If using local files, file system path to the binlog files is needed
+  --let $binlog_f1_full= $MYSQLD_DATADIR/$binlog_f1
+  --let $binlog_f2_full= $MYSQLD_DATADIR/$binlog_f2
+}
+
+--echo #
+--echo # --stop-position tests
+--echo #
+
+--echo #  Case 1.a) With one binlog file, a --stop-position before the end of
+--echo # the file should not result in a warning
+--echo # MYSQL_BINLOG $PARAM_READ_FROM_REMOTE_OUT --stop-position=binlog_f1_pre_rotate binlog_f1_full --result-file=$binlog_out_relpath 2>&1
+--exec $MYSQL_BINLOG $PARAM_READ_FROM_REMOTE --stop-position=$binlog_f1_pre_rotate $binlog_f1_full --result-file=$binlog_out 2>&1
+
+--echo #
+--echo #  Case 1.b) With one binlog file, a --stop-position at the exact end of
+--echo # the file should not result in a warning
+--echo # MYSQL_BINLOG $PARAM_READ_FROM_REMOTE_OUT --stop-position=binlog_f1_end binlog_f1_full --result-file=$binlog_out_relpath 2>&1
+--exec $MYSQL_BINLOG $PARAM_READ_FROM_REMOTE --stop-position=$binlog_f1_end $binlog_f1_full --result-file=$binlog_out 2>&1
+
+--echo #
+--echo #  Case 1.c) With one binlog file, a --stop-position past the end of the
+--echo # file should(!) result in a warning
+--let $binlog_f1_over_eof= `SELECT $binlog_f1_end + 10`
```

Comment:
Not sure why I used a value of `10` when I gave this to you... I think we should make this extra strict and change `10` to `1` to catch the warning exactly where it should be output.  (also on line `105`)

Reply:
Thanks for reviewing. Updated a new revision to address the comments. Please take a look.